### ISSUE TITLE
[Q] Wandbcallback keras models get overwritten

### ISSUE THREAD

**Issue by ValentinKoch** [wandb/wandb#3892]
Hi, im not sure how exactly this happens but my models that I am saving online keep getting overwritten by newer models if saving with wandbcallback, seems like its happening even between projects. My code looks roughly like this:

wandb_run=wandb.init(config=params,project="xxx")
wandb_callback=WandbCallback( log_weights=True)

 model.fit(.... ,callbacks=[wandb_callback])

when restoring model:

best_model = wandb.restore('model-best.h5', run_path="myname/xyz/t0yp5erz")
model.load_weights(best_model.name)

What am I doing wrong?

Thanks for helping!

**Comment by nate-wandb** [comment]
Hi @ValentinKoch,
Can you show how you are saving your model? Are you using `wandb.save()`? The `log_weights` argument of WandbCallback will log the histogram of the weights but this is not an actual .h5 model weights file. 

Thank you,
Nate

**Comment by ValentinKoch** [comment]
Ah yes, you are right. Actually its also done by WandbCallback, its default initialization has save_model=True, which I do not change. This saves the .h5 model file. 
Was just looking at the implementation of the WandbCallback class, maybe its that line:

 self.model.save(self.filepath, overwrite=True)

line 964 in https://github.com/wandb/client/blob/a339333b3ee93864daf416f04c1501186dffac5c/wandb/integration/keras/keras.py#L950

Would be very useful if we could change the overwrite to false or specify a model name to avoid that overwriting. Or you could make a warning when overwriting another model or so. I guess I should specify a different path each run in wandb.init() , maybe that was my bad. still think a model name would be nice to be able to chose :)

**Comment by MBakirWB** [comment]
Hi @ValentinKoch , following up on this. You are correct for WandbCallBack when saving a model, the existing model is replaced by setting `overwrite=True`. We overwrite the model when it's determined that the current model beats all previous epochs with respect to the monitored metric (i.e, `val_loss` ). We use the default argument value, `overwrite=True`, for  the Keras `save_model()` function. I do recognize that when overwriting we can give a user a warning or perhaps allow them to set `overwrite=False` which would trigger Keras to manually ask the user whether to overwrite or not. I will submit a feature request for engineering to review and provide an update when available.
